File: mobile/l9_back/internal/models/statistics.go
package models

type Statistics struct {
	TotalAmount  float64             `json:"totalAmount"`
	ExpenseCount int                 `json:"expenseCount"`
	ByCategory   []CategoryStat      `json:"byCategory"`
	ByPeriod     []PeriodStat        `json:"byPeriod"`
}

type CategoryStat struct {
	Category string  `json:"category"`
	Amount   float64 `json:"amount"`
	Count    int     `json:"count"`
}

type PeriodStat struct {
	Period string  `json:"period"`
	Amount float64 `json:"amount"`
}

File: mobile/l9_2/app/src/main/java/com/aemarkov/lab2/data/models/Statistics.kt
package com.aemarkov.lab2.data.models

data class Statistics(
    val totalAmount: Double,
    val expenseCount: Int,
    val byCategory: List<CategoryStatistics>,
    val byPeriod: List<PeriodStatistics>
)

data class CategoryStatistics(
    val category: String,
    val amount: Double,
    val count: Int
)

data class PeriodStatistics(
    val period: String,
    val amount: Double
)

